src/Agent.Plugins/GitSourceProvider.cs

```diff
@@ -288,6 +298,22 @@ public string GenerateAuthHeader(AgentTaskPluginExecutionContext executionContex
                     (repository.Endpoint.Id == Guid.Empty && string.Equals(x.Name, repository.Endpoint.Name.ToString(), StringComparison.OrdinalIgnoreCase)));
             }
 
+            bool isForkedBuild = StringUtil.ConvertToBoolean(executionContext.Variables.GetValueOrDefault("System.PullRequest.IsFork")?.Value);
+            executionContext.Debug($"isForkedBuild: {isForkedBuild}");
+            // use the SystemVssConnection for Azure Repos forked builds
+            if(endpoint == null && isForkedBuild && UseSystemVssConnectionForForkedBuilds())
+            {
+                endpoint = executionContext.Endpoints.Single();
```

Comment:
single will throw if there's more than one in the list. This doesn't seem correct.